addons/wowl/static/tests/webclient/web_client_integrated_tests.ts

```diff
@@ -5376,6 +5376,33 @@ QUnit.module("Action Manager Legacy Tests Porting", (hooks) => {
     webClient.destroy();
   });
 
+  QUnit.test("history back calls on_close handler of dialog action with 2 breadcrumbs", async function (assert) {
+    assert.expect(7);
+
+    const webClient = await createWebClient({ baseConfig });
+    await doAction(webClient, 1); // kanban
+    await doAction(webClient, 3); // list
+    assert.containsOnce(webClient.el!, '.o_list_view');
+
+    function onClose() {
+      assert.step("on_close");
+    }
+    // open a new dialog form
+    await doAction(webClient, 5, { onClose });
+    assert.containsOnce(webClient.el!, ".modal");
+    assert.containsOnce(webClient.el!, '.o_list_view');
+
+    const ev = new Event("history-back", { bubbles: true, cancelable: true });
+    webClient.el!.querySelector(".o_view_controller")!.dispatchEvent(ev);
```

Comment:
if you want to have a less 'artificial' test (depending less on the implementation details), I think that a valid flow would be to spawn an action in target new with a form view (only), and no res_id: it will open a form create mode in a dialog. Then, click on 'Discard' will do an history back (should close the dialog)